### PR TITLE
update to work with python up to 3.9, drop python2 support

### PR DESCRIPTION
**Pull request by gnmerritt**
This makes things work with `pyramid==2.0` and newer Python versions

Pyramid 2.0 drops Python 2 support, so I figure it's fine to do here too. They removed the `pyramid.compat` module. I hoisted uses up into this module since I'm in a bit of a hurry, things could probably be simplified even more.

### CODE REVIEW

File: pyramid_log/__init__.py [hunk 1]
```diff
@@ -53,6 +59,10 @@ class Formatter(logging.Formatter):
     determine the current request.
 
     '''
+    def __init__(self, *args, **kwargs):
+        kwargs['validate'] = False  # '.' not allowed in format strings by default
```
Comment (dairiki) [inline pyramid_log/__init__.py:63]: The `validate` parameter was introduced in python 3.8.  In earlier python's its use will result in a `TypeError: unexpected keyword argument`.

**Comment by dairiki** [comment]
Thank you!

I'm fine with dropping Python 2 support.
Given that, from a quick look, I concur that none of those uses `pyramid.compat` are necessary.

I'll look at that further in the next couple of days (if you don't get to it before me.)

**Comment by dairiki** [comment]
@gnmerritt I've just done some cleanup of your work and pushed the result to [dairiki:pyramid-2.0](https://github.com/dairiki/pyramid_log/commits/pyramid-2.0).

I plan to do a bit of work on modernizing the packaging in the master, then release that as 1.0 (terminus of pyramid-1.x support).
Then I will merge this work and release that as 2.0.

Comments welcome.

**Comment by gnmerritt** [comment]
Sounds good, thanks for being a maintainer!

**Comment by dairiki** [comment]
I've just released version [1.0](https://pypi.org/project/pyramid-log/1.0.0/) which should work with python 2.7, 3.6-3.10 and any version of pyramid.

I still plan on working on a 2.0 release which will drop support for py2.  That does simplify the code quite a bit.  (And maybe it will be not too hard to support the other format styles — `style='{'` and `style='$'` — which are supported by current versions of [`logging.Formatter`](https://docs.python.org/3/library/logging.html#logging.Formatter).)

**Pull request merged**